File: packages/maps-launcher/android/src/main/java/io/capawesome/capacitorjs/plugins/mapslauncher/classes/results/GetAvailableAppsResult.java
package io.capawesome.capacitorjs.plugins.mapslauncher.classes.results;

import androidx.annotation.NonNull;
import com.getcapacitor.JSArray;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.mapslauncher.interfaces.Result;
import java.util.List;

public class GetAvailableAppsResult implements Result {

    @NonNull
    private final List<String> apps;

    public GetAvailableAppsResult(@NonNull List<String> apps) {
        this.apps = apps;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSArray appsArray = new JSArray();
        for (String app : apps) {
            appsArray.put(app);
        }
        JSObject result = new JSObject();
        result.put("apps", appsArray);
        return result;
    }
}

File: packages/maps-launcher/ios/Plugin/Classes/Results/GetAvailableAppsResult.swift
import Foundation
import Capacitor

@objc public class GetAvailableAppsResult: NSObject, Result {
    let apps: [String]

    init(apps: [String]) {
        self.apps = apps
    }

    @objc public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["apps"] = apps
        return result as AnyObject
    }
}
